WordPress/Classes/ViewRelated/Tips/AppTips.swift

```diff
@@ -0,0 +1,79 @@
+import Foundation
+import SwiftUI
+import TipKit
+import Combine
+
+enum AppTips {
+    static func initialize() {
+        guard Feature.enabled(.tipKit), #available(iOS 17, *) else { return }
+        do {
+            try Tips.configure([.displayFrequency(.daily)])
+        } catch {
+            DDLogError("Error initializing tips: \(error)")
+        }
+    }
+
+    @available(iOS 17, *)
+    struct SitePickerTip: Tip {
+        let id = "site_picker_tip"
+
+        var title: Text {
+            Text(NSLocalizedString("tips.sitePickerTip.title", value: "Your Sites", comment: "Tip for site picker"))
+        }
+
+        var message: Text? {
+            Text(NSLocalizedString("tips.sitePickerTip.message", value: "Tap to select a different site or create a new one", comment: "Tip for site picker"))
+        }
+
+        var image: Image? {
+            Image(systemName: "rectangle.stack.badge.plus")
+        }
+
+        @Parameter(.transient)
+        static var blogCount: Int = 0
+
+        var rules: [Rule] {
+            #Rule(Self.$blogCount) { $0 > 1 }
+        }
+
+        var options: [any TipOption] {
+            MaxDisplayCount(1)
+        }
+    }
+}
+
+extension UIViewController {
+    /// Registers a popover to be displayed for the given tip.
+    @available(iOS 17, *)
+    func registerTipPopover(
```

Comment:
TipKit API for showing popovers was a bit too [low-level](https://developer.apple.com/documentation/tipkit/tipuipopoverviewcontroller), so I decided to add this convenience API on top of it.

Reply:
The term "register" gives of nothing will actually happen after this function is called. However, if I understand correctly, the function shows a tip if it certain criteria match. What do you think changing the name to something like `presentTipPopoverIfNeeded`?

Reply:
I'm open to suggestions, but `presentTipPopoverIfNeeded` sounds like a one-shot method. This method subscribes to a `tip.shouldDisplayUpdates` async sequences and shows/hides the popover as needed.

The SwiftUI [method](https://developer.apple.com/documentation/swiftui/view/popovertip(_:arrowedge:action:)) with the same functionality is called `popoverTip(_:arrowEdge:action:)`.

Reply:
I misunderstood how the `tip.shouldDisplayUpdates` API works. The function name makes more sense to me now.